src/status_im/contexts/wallet/events.cljs

```diff
@@ -547,3 +551,36 @@
  :wallet/process-watch-only-account-from-backup
  (fn [_ [{:keys [backedUpWatchOnlyAccount]}]]
    {:fx [[:dispatch [:wallet/process-account-from-signal backedUpWatchOnlyAccount]]]}))
+
+(defn process-keypairs
+  [{:keys [db]} [keypairs]]
+  (let [existing-keypairs-by-id               (get-in db [:wallet :keypairs])
+        existing-accounts-by-address          (get-in db [:wallet :accounts])
+        {:keys [removed-keypair-ids
+                removed-account-addresses
+                updated-keypairs-by-id
+                updated-accounts-by-address]} (data-store/process-keypairs keypairs)
+        updated-keypair-ids                   (set (keys updated-keypairs-by-id))
+        updated-account-addresses             (set (keys updated-accounts-by-address))
+        existing-account-addresses            (set (keys existing-accounts-by-address))
+        new-account-addresses                 (clojure.set/difference updated-account-addresses
+                                                                      existing-account-addresses)
+        old-account-addresses                 (->> (vals existing-accounts-by-address)
+                                                   (filter (fn [{:keys [address key-uid]}]
+                                                             (and (contains? updated-keypair-ids key-uid)
+                                                                  (nil? (get updated-accounts-by-address
```

Comment:
Just thinking out loud :)

Could you explain why we need to check for `nil?` explicitly and not just `(contains? updated-accounts-by-address address)`. Would it be possible to make sure `updated-accounts-by-address` don't contain keys with nil values and then rely on the semantics of `contains?`?

Reply:
Thanks for spotting this 🙏 

For some reason I thought that `contains?` wouldn't work correctly on the map `updated-accounts-by-address`, but I think you're right that `contains?` will check for the presence of the key. Fortunately the code does not depend on any explicit `nil` (I think), so it should be an easy change 👍